### PR TITLE
Jsxc instead of jwchat

### PR DESCRIPTION
**Pull request by SunilMohanAdapa**
Use Javascript XMPP Client instead of jwchat.

### CODE REVIEW

File: plinth/modules/xmpp/__init__.py [hunk 1]
```diff
@@ -39,16 +39,15 @@
 
 managed_services = ['ejabberd']
 
-managed_packages = ['jwchat', 'ejabberd']
+managed_packages = ['libjs-jsxc', 'ejabberd']
```
Comment (jvalleroy) [inline plinth/modules/xmpp/__init__.py]: I think we will need to use setup() to upgrade existing systems.

**Comment by SunilMohanAdapa** [comment]
On 09/18/2016 07:51 PM, James Valleroy wrote:
> *@jvalleroy* commented on this pull request.
> 
> ------------------------------------------------------------------------
> 
> In plinth/modules/xmpp/__init__.py
> <https://github.com/freedombox/Plinth/pull/607#pullrequestreview-476153>:
> 
>> @@ -39,16 +39,15 @@
>  
>  managed_services = ['ejabberd']
>  
> -managed_packages = ['jwchat', 'ejabberd']
> +managed_packages = ['libjs-jsxc', 'ejabberd']
> 
> I think we will need to use setup() to upgrade existing systems.
> 

I had written a lot of code for upgrade and later removed it leading to
this error.  I believe a simple increment of the version number should
take care of this.

-- 
Sunil

**Comment by jvalleroy** [comment]
Testing this in vagrant now, but I'm seeing two 404's when the page loads:
- https://localhost:4430/plinth/static/xmpp/libjs-jsxc/jsxc.css
- https://localhost:4430/plinth/static/xmpp/libjs-jsxc/lib/otr/build/dsa-webworker.js

But jsxc.css is installed in /usr/share/libjs-jsxc/ and symlinked in /vagrant/plinth/modules/xmpp/static/libjs-jsxc/. And dsa-webworker.js is installed in /usr/share/javascript/jsxc/lib/otr/lib/. Any idea what would cause this?

**Comment by SunilMohanAdapa** [comment]
On 09/18/2016 09:12 PM, James Valleroy wrote:
> Testing this in vagrant now, but I'm seeing two 404's when the page loads:
> 
>   * https://localhost:4430/plinth/static/xmpp/libjs-jsxc/jsxc.css
>   * https://localhost:4430/plinth/static/xmpp/libjs-jsxc/lib/otr/build/dsa-webworker.js
> 
> But jsxc.css is installed in /usr/share/libjs-jsxc/ and symlinked in
> /vagrant/plinth/modules/xmpp/static/libjs-jsxc/. And dsa-webworker.js is
> installed in /usr/share/javascript/jsxc/lib/otr/lib/. Any idea what
> would cause this?

The symlinks I have created in source are not being created by our
setup.py in the target directory.  If you run the code as ./run, it
should work.

We need to write special code in ./setup.py to create symlinks or handle
this in Debian package (which will anyway crib about unsafe absolute
symlinks in source code).

This mess is due to https://bugs.debian.org/838183 . Once this is fixed
can remove the hacks.

-- 
Sunil

**Comment by jvalleroy** [comment]
I merged this after incrementing the version number. I think we can handle the symlink issue in the Debian package.

**Pull request closed without merging**